### PR TITLE
dev/core#527 Permit translation of Public Title field for Profiles

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes the field 'Public Title' to be translatable on the profile form

Before
----------------------------------------
Not translatable

After
----------------------------------------
screenshot 2018-11-16 12 05 46

Technical Details
----------------------------------------
In order to fix this field I have partially converted the form to an EntityForm. To finish the form we probably need to figure out how to deal with the accordian in it - but I did look at one challenge - which was having a way to add some text after the html & before the <br> which I did in the tpl

Comments
----------------------------------------
@mattwire ping
@samuelsov this fixes one of the fields you describe in https://lab.civicrm.org/dev/core/issues/527
Note that the next field on the display is 'description' and that doesn't seem to be localised in the schema yet. I think there are some upgrade challenges in changing that

### CODE REVIEW

File: CRM/UF/Form/Group.php [hunk 1]
```diff
@@ -29,15 +29,52 @@
...
+   * Set entity fields to be assigned to the form.
+   */
+  protected function setEntityFields() {
+    $this->entityFields = [
+      'title' => ['name' => 'title'],
+      'frontend_title' => ['name' => 'frontend_title'],
+      'description' => ['name' => 'description', 'help' => ['id' => 'id-description', 'file' => 'CRM/UF/Form/Group.hlp']],
+      'uf_group_type' => ['name' => 'uf_group_type', 'not-auto-addable' => TRUE, 'help' => ['id' => 'id-used_for', 'file' => 'CRM/UF/Form/Group.hlp'], 'post_html_text' => ' ' . $this->getOtherModuleString()]
+    ];
+  }
```
Comment (mattwire) [inline CRM/UF/Form/Group.php]: Yes.

File: CRM/UF/Form/Group.php [hunk 2]
```diff
@@ -237,17 +267,6 @@ public function setDefaultValues() {
...
-      //get the uf join records for current uf group other than default modules
-      $otherModules = array();
-      $otherModules = CRM_Core_BAO_UFGroup::getUFJoinRecord($this->_id, TRUE, TRUE);
-      if (!empty($otherModules)) {
-        $otherModuleString = NULL;
-        foreach ($otherModules as $key) {
-          $otherModuleString .= " [ x ] <label>" . $key . "</label>";
-        }
-        $this->assign('otherModuleString', $otherModuleString);
-      }
```
Comment (mattwire) [inline CRM/UF/Form/Group.php]: Yes. Moved to it's own function `getOtherModuleString()`

**Review by mattwire — APPROVED** [review]
@eileenmcnaughton I've r-run this and done a code review.  It works perfectly.

I have not tested this on a multilingual install and I don't quite understand how this makes the field translateable but I'm assuming that's part of the magic of entityform?

**Comment by eileenmcnaughton** [comment]
merging based on @mattwire review

@mattwire yes the field level template for entity form looks at is_localizable & renders translation box if appropriate

File: templates/CRM/Core/Form/Field.tpl [hunk 3]
```diff
@@ -33,7 +33,7 @@
     {/capture}{help id=$help.id file=$help.file}{/if}
     {if $action == 2 && $fieldSpec.is_add_translate_dialog}{include file='CRM/Core/I18n/Dialog.tpl' table=$entityTable field=$fieldName id=$entityID}{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Core/Form/Field.tpl]: @mattwire this line ^^

**Pull request merged**